plugins/ccgx/fu-ccgx-dmc-device.c

```diff
@@ -0,0 +1,703 @@
+	if (!g_usb_device_control_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+					    G_USB_DEVICE_DIRECTION_DEVICE_TO_HOST,
+					    G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					    G_USB_DEVICE_RECIPIENT_DEVICE,
+					    DMC_RQT_CODE_DOCK_STATUS, /* request */
+					    0, /* value */
+					    0, /* index */
+					    (guint8*)dock_status, /* data */
+					    DMC_GET_STATUS_MIN_LEN,  /* length */
+					    NULL, /* actual length */
+					    DMC_CONTROL_TRANSFER_DEFAULT_TIMEOUT, /* timeout */
+					    NULL, error)) {
+		g_prefix_error (error, "get_dock_status min size error:");
+		return FALSE;
+	}
+	if (dock_status->status_length <= sizeof(DmcDockStatus)) {
+		/* read full status lenght */
+		if (!g_usb_device_control_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+						    G_USB_DEVICE_DIRECTION_DEVICE_TO_HOST,
+						    G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+						    G_USB_DEVICE_RECIPIENT_DEVICE,
+						    DMC_RQT_CODE_DOCK_STATUS, /* request */
+						    0, /* value */
+						    0, /* index */
+						    (guint8*)dock_status, /* data */
+						    sizeof(DmcDockStatus),  /* length */
+						    NULL, /* actual length */
+						    DMC_CONTROL_TRANSFER_DEFAULT_TIMEOUT, /* timeout */
+						    NULL, error)) {
+			g_prefix_error (error, "get_dock_status actual size error:");
+			return FALSE;
+		}
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_ccgx_dmc_device_send_reset_state_machine (FuCcgxDmcDevice *self, GError **error)
+{
+	if (!g_usb_device_control_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+					    G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					    G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					    G_USB_DEVICE_RECIPIENT_DEVICE,
+					    DMC_RQT_CODE_RESET_STATE_MACHINE, /* request */
+					    0, /* value */
+					    0, /* index */
+					    0, /* data */
+					    0,  /* length */
+					    NULL, /* actual length */
+					    DMC_CONTROL_TRANSFER_DEFAULT_TIMEOUT, /* timeout */
+					    NULL, error)) {
+		g_prefix_error (error, "send reset state machine error:");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_ccgx_dmc_device_send_sort_reset (FuCcgxDmcDevice *self,
+				    gboolean reset_later,
+				    GError **error)
+{
+	if (!g_usb_device_control_transfer (fu_usb_device_get_dev (FU_USB_DEVICE (self)),
+					    G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					    G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					    G_USB_DEVICE_RECIPIENT_DEVICE,
+					    DMC_RQT_CODE_SOFT_RESET, /* request */
+					    reset_later, /* value */
+					    0, /* index */
+					    0, /* data */
+                                            0,  /* length */
+                                            NULL, /* actual length */
+					    DMC_CONTROL_TRANSFER_DEFAULT_TIMEOUT, /* timeout */
+					    NULL, error)) {
+		g_prefix_error (error, "send reset error:");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+
+static gboolean
+fu_ccgx_dmc_device_send_start_upgrade (FuCcgxDmcDevice *self,
+				       guint8* custom_meta_data,
```

Comment:
the `*` on the name please

Reply:
Fixed it on today commit (ccgx: fix code alignment, modify code for review)